tests/components/reolink/conftest.py

```diff
@@ -0,0 +1,70 @@
+"""Setup the Reolink tests."""
+from collections.abc import Generator
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+TEST_HOST = "1.2.3.4"
+TEST_HOST2 = "4.5.6.7"
+TEST_USERNAME = "admin"
+TEST_USERNAME2 = "username"
+TEST_PASSWORD = "password"
+TEST_PASSWORD2 = "new_password"
+TEST_MAC = "ab:cd:ef:gh:ij:kl"
+TEST_PORT = 1234
+TEST_NVR_NAME = "test_reolink_name"
+TEST_USE_HTTPS = True
+
+
+def get_mock_info(error=None):
+    """Return a mock gateway info instance."""
+    host_mock = Mock()
```

Comment:
Why not patch the client library and set autospec to `True`?

Reply:
Sorry, I am a bit new to this and am a bit stuck:

I am trying to do this, and have now succesfully patched the client library with autospec True in a fixture.
But how do I now overwrite one of the funcions in that client library to have a side-effect?

so:
```
@pytest.fixture(name="reolink_connect")
def reolink_connect_fixture(mock_get_source_ip):
    """Mock reolink connection."""
    with patch(
        "homeassistant.components.reolink.host.webhook.async_register",
        return_value=True,
    ), patch("homeassistant.components.reolink.host.Host", autospec=True) as host_mock_class:
        host_mock = host_mock_class.return_value
        host_mock.get_host_data = AsyncMock(return_value=None)
        host_mock.get_states = AsyncMock(return_value=None)
............................
        host_mock.renewtimer = 600
        yield host_mock
```

And now I want to overwrite the `get_host_data` function to throw a error, someting like:
`with patch("homeassistant.components.reolink.host.Host.get_host_data", AsyncMock(side_effect=ReolinkError("Test error"))):`

But that does not work, how am I supposed to do this?

Reply:
Never mind, I figured it out.

Reply:
Done